src/lib.rs

```diff
@@ -183,10 +186,34 @@ pub struct Upwards;
 #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd)]
 pub struct Downwards;
 
+#[doc(hidden)]
+pub trait Direction<Idx> {
```

Comment:
Do we want other crates to be able to implement this? Even though this is hidden in the docs, in theory, they can still use this. If we want to prevent this, we might want to seal this trait (cf. <https://rust-lang-nursery.github.io/api-guidelines/future-proofing.html>).